### ISSUE TITLE
Reducing load on GPU to avoid CUDA errors

### ISSUE THREAD

**Issue by mikeyEcology** [open-mmlab/mmaction2#462]
Does anyone have suggestions for reducing the load on the GPUs and allowing training to run longer with less work on the GPUS? I keep getting errors like this:

> RuntimeError: CUDA error: CUBLAS_STATUS_INVALID_VALUE when calling cublasSetMathMode(handle, CUBLAS_TF32_TENSOR_OP_MATH)

when trying to train a TSN recognition model. I'm setting OMP_NUM_THREADS=1 and using a reduced batch size. I've also set workers_per_gpu=0, but I still get this error when using two gpus or one gpu. I'm using the new RTX 3090 cards.
Do folks have other general suggestions for avoiding these errors? I've pasted my config below, for reference
```
model = dict(
    type='Recognizer2D',
    backbone=dict(
        type='ResNet',
        pretrained='torchvision://resnet50',
        depth=50,
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        norm_eval=True),
    cls_head=dict(
        type='TSNHead',
        num_classes=3,
        in_channels=2048,
        spatial_type='avg',
        consensus=dict(type='AvgConsensus', dim=1),
        dropout_ratio=0.5,
        init_std=0.001))
# model training and testing settings
train_cfg = None
test_cfg = dict(average_clips=None)
# pipeline
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_bgr=False)
train_pipeline = [
    dict(type='SampleFrames', clip_len=1, frame_interval=1, num_clips=4),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    # dict(
    #     type='MultiScaleCrop',
    #     input_size=224,
    #     scales=(1, 0.875, 0.75, 0.66),
    #     random_crop=False,
    #     max_wh_scale_gap=1,
    #     num_fixed_crops=13),
    dict(type='Resize', scale=(224, 224), keep_ratio=False),
    dict(type='Flip', flip_ratio=0.5),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
val_pipeline = [
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=4,
        test_mode=True),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
test_pipeline = [
    dict(
        clip_len=1,
        frame_interval=1,
        num_clips=4,
        test_mode=True),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
# dataset settings
dataset_type = 'RawframeDataset'
data_root = '/home/user/projects/databattransects_2020/data/ava_format/data/videos_jpg_7frames/'
data_root_val = '/home/user/projects/databattransects_2020/data/ava_format/data/videos_jpg_7frames/'
ann_file_train = '/home/user/projects/databattransects_2020/data/ava_format/data/annotations/annotations_7frame_train.txt'
ann_file_val = '/home/user/projects/databattransects_2020/data/ava_format/data/annotations/annotations_7frame_val.txt'
ann_file_test = '/home/user/projects/databattransects_2020/data/ava_format/data/annotations/annotations_7frame_val.txt'
data = dict(
    videos_per_gpu=2,
    workers_per_gpu=0,
    # add these dataloader params
    train_dataloader=dict(drop_last=True),
    val_dataloader=dict(videos_per_gpu=1),
    test_dataloader=dict(videos_per_gpu=1),
    train=dict(
        type=dataset_type,
        ann_file=ann_file_train,
        data_prefix=data_root,
        filename_tmpl='frame_{:06}.jpg',
        pipeline=train_pipeline
        ),
    val=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        data_prefix=data_root_val,
        filename_tmpl='frame_{:06}.jpg',
        pipeline=val_pipeline
        ),
    test=dict(
        type=dataset_type,
        ann_file=ann_file_test,
        data_prefix=data_root_val,
        filename_tmpl='frame_{:06}.jpg',
        pipeline=test_pipeline
        ))
dataset_A_train = dict(
    type='MyDataset',
    ann_file = ann_file_train
)

# optimizer
optimizer = dict(
    type='Adam', lr=0.001, weight_decay=0.0001)  # this lr is used for 1 gpus
    #type='SGD', lr=0.01, momentum=0.9,
    #weight_decay=0.0005)  # this lr is used for 8 gpus
optimizer_config = dict(grad_clip=None) #dict(grad_clip=dict(max_norm=20, norm_type=2))
# learning policy
lr_config = dict(policy='step', step=7) #dict(policy='step', step=[20, 40])
total_epochs = 20
checkpoint_config = dict(interval=1)
evaluation = dict(
    interval=2, metrics=['top_k_accuracy', 'mean_class_accuracy'])
log_config = dict(
    interval=50, #20,
    hooks=[
        dict(type='TextLoggerHook'),
        # dict(type='TensorboardLoggerHook'),
    ])
# runtime settings
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = '/home/user/repos/mmaction2/work_dirs/tsn/'
load_from = None
resume_from = None
workflow = [('train', 1)]
```
Thank you

More detailed error message:

```
2020-12-18 18:18:56,574 - mmaction - INFO - These parameters in pretrained checkpoint are not loaded: {'fc.bias', 'fc.weight'}
2020-12-18 18:18:56,663 - mmaction - INFO - Start running, host: usr@lar-excelsior-srv.local, work_dir: /home/usr/repos/mmaction2/work_dirs/tsn
2020-12-18 18:18:56,664 - mmaction - INFO - workflow: [('train', 1)], max: 20 epochs
Traceback (most recent call last):
  File "./tools/train.py", line 178, in <module>
    main()
  File "./tools/train.py", line 174, in main
    meta=meta)
  File "/home/usr/repos/mmaction2/mmaction/apis/train.py", line 141, in train_model
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs, **runner_kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 125, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 50, in train
    self.run_iter(data_batch, train_mode=True)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 30, in run_iter
    **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/parallel/distributed.py", line 36, in train_step
    output = self.module.train_step(*inputs[0], **kwargs[0])
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/base.py", line 207, in train_step
    losses = self(imgs, label, return_loss=True, **aux_info)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/base.py", line 169, in forward
    return self.forward_train(imgs, label, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/recognizer2d.py", line 17, in forward_train
    x = self.extract_feat(imgs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/fp16_utils.py", line 84, in new_func
    return old_func(*args, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/base.py", line 72, in extract_feat
    x = self.backbone(imgs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/backbones/resnet.py", line 536, in forward
    x = self.conv1(x)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/cnn/bricks/conv_module.py", line 192, in forward
    x = self.conv(x)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/cnn/bricks/wrappers.py", line 58, in forward
    return super().forward(x)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 394, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 391, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: CUDA error: CUBLAS_STATUS_INVALID_VALUE when calling `cublasSetMathMode(handle, CUBLAS_TF32_TENSOR_OP_MATH)`
Traceback (most recent call last):
  File "./tools/train.py", line 178, in <module>
    main()
  File "./tools/train.py", line 174, in main
    meta=meta)
  File "/home/usr/repos/mmaction2/mmaction/apis/train.py", line 141, in train_model
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs, **runner_kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 125, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 50, in train
    self.run_iter(data_batch, train_mode=True)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 30, in run_iter
    **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/parallel/distributed.py", line 36, in train_step
    output = self.module.train_step(*inputs[0], **kwargs[0])
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/base.py", line 207, in train_step
    losses = self(imgs, label, return_loss=True, **aux_info)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/base.py", line 169, in forward
    return self.forward_train(imgs, label, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/recognizer2d.py", line 17, in forward_train
    x = self.extract_feat(imgs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/runner/fp16_utils.py", line 84, in new_func
    return old_func(*args, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/recognizers/base.py", line 72, in extract_feat
    x = self.backbone(imgs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/repos/mmaction2/mmaction/models/backbones/resnet.py", line 536, in forward
    x = self.conv1(x)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/cnn/bricks/conv_module.py", line 192, in forward
    x = self.conv(x)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/cnn/bricks/wrappers.py", line 58, in forward
    return super().forward(x)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 394, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 391, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: CUDA error: CUBLAS_STATUS_INVALID_VALUE when calling `cublasSetMathMode(handle, CUBLAS_TF32_TENSOR_OP_MATH)`
Traceback (most recent call last):
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/distributed/launch.py", line 306, in <module>
    main()
  File "/home/usr/bin/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/distributed/launch.py", line 298, in main
    cmd=cmd)
subprocess.CalledProcessError: Command '['/home/usr/bin/anaconda3/envs/open-mmlab/bin/python', '-u', './tools/train.py', '--local_rank=1', '/home/usr/repos/mmaction2/configs/recognition/tsn/data_config1.py', '--launcher', 'pytorch']' returned non-zero exit status 1.
```

**Comment by innerlee** [comment]
Run the example in python repl
```python
>>> m = nn.Conv2d(16, 33, (3, 5), stride=(2, 1), padding=(4, 2), dilation=(3, 1))
>>> input = torch.randn(20, 16, 50, 100)
>>> output = m(input)
```

**Comment by mikeyEcology** [comment]
This gives the same error. Clearly the problem is not with mmaction2. This is strange because I've run other `torch.nn` models on this server. 
Where did you find this test example and do you have any recommendations? 
Thank you!

**Comment by mikeyEcology** [comment]
I managed to solve this problem by making a small modification to the [installation instructions](https://github.com/open-mmlab/mmaction2/blob/master/docs/install.md). After step (a), creating and activating environment. I ran `conda install pytest`. This is required for the mmcv-full to load, but if you install it after installing pytorch it will change the pytorch version, and cause problems (at least on my machine).
